### PR TITLE
Implementation of EAD attack

### PR DESCRIPTION
**Pull request by FlashTek**
Implementation of the EAD attack described in the paper [EAD: Elastic-Net Attacks to Deep Neural Networks via Adversarial Examples](https://arxiv.org/abs/1709.04114)

### CODE REVIEW

**Comment by jonasrauber** [comment]
I've fixed the tests on master. You can now update this PR (e.g. by rebasing on or merging master).

File: foolbox/attacks/ead.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+import numpy as np
+import logging
+
+from .base import Attack
+from .base import call_decorator
+from ..utils import onehot_like
+
+
+class EADAttack(Attack):
+    """""Gradient based attack which uses an elastic-net regularization [1].
```
Comment (jonasrauber) [inline foolbox/attacks/ead.py]: looks like this are too many `""""""` ?

**Comment by jonasrauber** [comment]
Thanks @FlashTek. Looks good. Can you look at one comment I made, see above, before I merge it. I guess the code is basically a modification of the CW attack code, right?

**Comment by jonasrauber** [comment]
Oh and could add it as a batch attack (there is already a batch-version of CW, so you can just look at the diff for CW). Doesn't need to be part of this PR.

**Comment by FlashTek** [comment]
@jonasrauber This PR already includes the batch-version of the EAD attack ([batch_attacks/ead.py](https://github.com/bethgelab/foolbox/pull/336/files#diff-3370a95d19711540b525db90eda87e6a)).

File: foolbox/attacks/ead.py (hunk 1 above)
Comment (FlashTek) [reply foolbox/attacks/ead.py]: Fixed in [a3720d6](https://github.com/bethgelab/foolbox/pull/336/commits/a3720d66efdcd09a16c242dabc2e2fd0b4d5c15e)

**Comment by jonasrauber** [comment]
Thanks for the fix and sorry for overlooking the batched version.

**Pull request merged**